e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/pdc-reconcile/mark-report-as-done.spec.js

```diff
@@ -0,0 +1,234 @@
+import pages from '../../../pages';
+import USERS from '../../../../fixtures/users';
+import { PDC_TEAMS } from '../../../../fixtures/teams';
+import { toIsoMonthStamp } from '../../../../support/utils/dateHelpers';
+import { NODE_TASKS } from '../../../../../../e2e-fixtures';
+import { UTILISATION_REPORT_RECONCILIATION_STATUS, createMockUtilisationReportDetails } from '../../../../fixtures/create-mock-utilisation-report-details';
+
+context(`${PDC_TEAMS.PDC_RECONCILE} users can mark reports as done and not done`, () => {
+  const submissionMonth = toIsoMonthStamp(new Date());
+  const utilisationReportDetailsAlias = 'utilisationReportDetailsAlias';
+
+  const displayStatusSelector = 'td > strong[data-cy="utilisation-report-reconciliation-status"]';
+  const tableCellCheckboxSelector = (reportId) => `th > div > div > input[data-cy="table-cell-checkbox--set-status--reportId-${reportId}"]`;
+
+  const statusWithBankId = [
+    {
+      bankId: undefined,
+      status: UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED,
+    },
+    {
+      bankId: undefined,
+      status: UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION,
+    },
+    {
+      bankId: undefined,
+      status: UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_COMPLETED,
+    },
+  ];
+
+  const getDisplayStatus = (utilisationReportReconciliationStatus) => {
+    switch (utilisationReportReconciliationStatus) {
+      case UTILISATION_REPORT_RECONCILIATION_STATUS.REPORT_NOT_RECEIVED:
+        return 'Not received';
+      case UTILISATION_REPORT_RECONCILIATION_STATUS.PENDING_RECONCILIATION:
+        return 'Pending reconciliation';
+      case UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_IN_PROGRESS:
+        return 'Reconciliation in progress';
+      case UTILISATION_REPORT_RECONCILIATION_STATUS.RECONCILIATION_COMPLETED:
+        return 'Report completed';
+      default:
+        throw new Error('Unrecognised utilisation report reconciliation status');
+    }
+  };
+
+  beforeEach(() => {
+    const visibleBanks = [];
+    cy.getAllBanks().then((getAllBanksResult) => {
+      getAllBanksResult
+        .filter((bank) => bank.isVisibleInTfmUtilisationReports)
+        .forEach((bank) => {
+          visibleBanks.push(bank);
+        });
+      cy.wrap(visibleBanks).its('length').should('be.gte', 3);
+    });
+
+    const mockUtilisationReportDetailsWithoutId = [];
+    cy.wrap(visibleBanks).each((bank, index) => {
```

Comment:
You need the `wrap` here as `map` will execute before `visibleBanks` has been populated, so `visibleBanks` is actually equal to `[]`. By `wrapping` it, you force cypress to wait for the previous cypress command to finish executing which, in this case, is the command which populates `visibleBanks`